flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/RandomAccessInputView.java

```diff
@@ -45,7 +45,12 @@ public RandomAccessInputView(ArrayList<MemorySegment> segments, int segmentSize)
 	{
 		this(segments, segmentSize, segmentSize);
 	}
-	
+
+	public RandomAccessInputView(ArrayList<MemorySegment> segments, int segmentSize, boolean dummy)
```

Comment:
This constructor is called twice from the constructor of the `ReduceHashTable`. Once to initialize the input view of the `RecordArea` and once for the `StagingArea`. Both areas will need at least one buffer. Maybe I am wrong, but if we give both one initial buffer, we do not need this additional constructor with the `dummy` flag. 

As a second alternative to the constructor with the `dummy` flag, we could also implement a constructor without the `ArrayList<MemorySegment>`, create it in the constructor, and add a `getSegementList()` method to access the created list.

What do you think, @ggevay?